bench-vortex/src/bin/tpch.rs

```diff
@@ -204,133 +310,133 @@ async fn bench_main(
-            {
-                continue;
-            }
-
-            if exclude_queries
-                .as_ref()
-                .is_some_and(|excluded| excluded.contains(&query_idx))
-            {
-                continue;
-            }
+                .is_none_or(|included| included.contains(query_idx))
+                && exclude_queries
+                    .as_ref()
+                    .is_none_or(|excluded| !excluded.contains(query_idx))
+        })
+        .collect();
+
+    for engine in engines {
+        for format in formats.iter().copied() {
+            match engine {
+                Engine::DataFusion => {
+                    let ctx = load_datasets(&url, format, emulate_object_store)
+                        .await
+                        .unwrap();
+
+                    let mut plans = Vec::new();
+
+                    for (query_idx, sql_queries) in tpch_queries.clone() {
+                        // Run benchmark as an async function
+                        let (fastest_result, plan) =
+                            benchmark_datafusion_query(query_idx, &sql_queries, iterations, &ctx)
+                                .await;
+
+                        // Row count verification
+                        let first_row_count = run_tpch_query(&ctx, &sql_queries, query_idx).await.0;
+                        row_counts.push((query_idx, format, first_row_count));
+
+                        // gather metrics
+                        for (idx, metrics_set) in VortexMetricsFinder::find_all(plan.as_ref())
+                            .into_iter()
+                            .enumerate()
+                        {
+                            metrics.merge_all_with_label(
+                                metrics_set,
+                                &[
+                                    Label::new("query_idx", query_idx.to_string()),
+                                    Label::new("vortex_exec_idx", idx.to_string()),
+                                ],
+                            );
+                        }
+                        plans.push((query_idx, plan.clone()));
+
+                        let storage = match url.scheme() {
+                            "s3" => "s3",
+                            "gcs" => "gcs",
+                            "file" => "nvme",
+                            otherwise => {
+                                warn!("unknown URL scheme: {}", otherwise);
+                                return ExitCode::FAILURE;
+                            }
+                        }
+                        .to_owned();
+
+                        measurements.push(QueryMeasurement {
+                            query_idx,
+                            engine: Engine::DataFusion,
+                            storage,
+                            time: fastest_result,
+                            format,
+                            dataset: "tpch".to_owned(),
```

Comment:
maybe a const or a enum

Reply:
Wasn't introduced as part of this PR, just moved. But agreed that it could be nice to do this diff.